### ISSUE TITLE
Properties order v1 -> v2

### ISSUE THREAD

**Issue by vitalik** [pydantic/pydantic#6209]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2 installed directly from the `main` branch, or equivalent

### Description

related https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6239747

```Python
class Payload(BaseModel):
    username: Optional[str] = None
    password: Optional[str] = None
```

when I generate schema the order of the fields changes:

```
{
  "properties": {
    "password": {
         ...
    },
    "username": {
         ...
    }
  },
  "title": "Payload",
  "type": "object"
}
```
(first goes password, instead of username)

this was working fine on v1

### Python, Pydantic & OS Version

```Text
pydantic version: 2.0b3
        pydantic-core version: 0.39.0 release build profile
                 install path: /private/tmp/fastapitest/.venv/lib/python3.9/site-packages/pydantic
               python version: 3.9.15 (main, Dec  6 2022, 21:16:57)  [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-13.4-arm64-arm-64bit
     optional deps. installed: ['typing-extensions']
```

**Comment by Kludex** [comment]
I don't think it's a bug per say, since it was introduced on purpose on #6043, but since we don't provide a way to customize it, I think we can revert it.

What do you think @adriangb ?

**Comment by vitalik** [comment]
> I don't think it's a bug per say

Yeah, technically it is not , but from UI perspective it's a nightmare :D

**Comment by lig** [comment]
I think that the intent in #6043 was to be deterministic and decouple the attributes order in JSON Schema from internal implementation changes. IMO, this doesn't contradict preserving user-defined attributes order.
